### PR TITLE
MRC-4244: Bug fix for Showing an error on a submitted rate

### PR DESCRIPTION
**Pull request by pearl-truss**
## Summary
To replicate:

1. Create a new submission

2. Select "contract action and rate certification" and fill out form

3. On the Rate details page link a rate certification that was submitted prior to when the rate name enhancement feature went live.
4. Click continue and fill out the rest of the submission
5. When you arrive at the Review and submit page...

Current behavior:
The Rate details will flag an error validation on the field  "Rates this rate certification covers" and prevent a submission, despite the rate being locked. 

Expected behavior:

We will not flag an error validation on a locked linked rate.

Displays the historic 'Programs this rate covers' if the linked rate is locked and historic

#### Screenshots

**BEFORE**
Screenshot 2024-07-03 at 10 02 45 AM

**AFTER**
Screenshot 2024-07-03 at 10 57 09 AM

#### Test cases covered

- Displays the old field for historic rates
- Renders the new field on unlocked rate

## QA guidance

I pulled down prod data to test this locally

### CODE REVIEW

File: services/app-web/src/pages/StateSubmission/ReviewSubmit/V2/ReviewSubmit/RateDetailsSummarySectionV2.tsx [hunk 1]
```diff
@@ -294,29 +294,47 @@ export const RateDetailsSummarySectionV2 = ({
...
-                                      <DataDetail
-                                          id="ratePrograms"
-                                          label="Rates this rate certification covers"
-                                          explainMissingData={
-                                              explainMissingData
-                                          }
-                                          children={ratePrograms(rate, false)}
-                                      />
+                                          isSubmittedOrCMSUser) ||
+                                          (rateFormData.deprecatedRateProgramIDs
```
Comment (haworku) [inline services/app-web/src/pages/StateSubmission/ReviewSubmit/V2/ReviewSubmit/RateDetailsSummarySectionV2.tsx:300]: nit: this feels hard to grok with so many chained conditionals nested in the jsx code - could we make this a variable higher up similar to `isSubmittedOrCMSUser` ? I think this is `isLockedAndOnlyHasHistoricRates` or something like that

File: services/app-web/src/pages/StateSubmission/ReviewSubmit/V2/ReviewSubmit/RateDetailsSummarySectionV2.test.tsx [hunk 2]
```diff
@@ -400,6 +401,75 @@ describe('RateDetailsSummarySection', () => {
...
+        expect(
+            screen.queryByRole('definition', {
+                name: 'Rates this rate certification covers',
+            })
+        ).not.toBeInTheDocument()
+    })
+
+    it('does not render the deprecated rate programs when present on an unlocked historic linked rate', async () => {
+        const statePrograms = mockMNState().programs
+        const contract = mockContractPackageSubmitted()
```
Comment (haworku) [inline services/app-web/src/pages/StateSubmission/ReviewSubmit/V2/ReviewSubmit/RateDetailsSummarySectionV2.test.tsx:446]: Where is there an "unlocked historic linked rate" here? I"m not seeing it in the test data but let me know what I'm missing

**Review by haworku — APPROVED** [review]
Bugfix looks good, had a couple comments but nothing blocking

File: services/app-web/src/pages/StateSubmission/ReviewSubmit/V2/ReviewSubmit/RateDetailsSummarySectionV2.test.tsx (hunk 2 above)
Comment (pearl-truss) [reply services/app-web/src/pages/StateSubmission/ReviewSubmit/V2/ReviewSubmit/RateDetailsSummarySectionV2.test.tsx:446]: Sorry that's a typo in the description

**Pull request merged**